src/MQTT.h

```diff
@@ -192,7 +190,7 @@ void writeSysParamsToMQTT(void) {
             }
             
             for (const auto& pair : mqttSensors) {
-                mqtt_publish(pair.first, number2string(pair.second()));
+                mqtt_publish(pair.first, pair.second());
```

Comment:
Why this change? it's shorter to just convert the numbers here rather than for each mqttSensors entry

Reply:
Or even just add an overload for mqtt_publish that accepts doubles

Reply:
> Or even just add an overload for mqtt_publish that accepts doubles

That would be nice. But this would require a modification of the value type of the `mqttSensors` map. The function has one specific return type, but next to the `double` values, `machineStateString` topic needs to return a `char *`. At the moment I do not have another solution for it.

Reply:
I'm not sure I understand what you mean. If you add an overload of mqtt_publish that accepts a double (that can then call number2string), you can call it with pair.second() here without number2string and just leave the type of mqttSensors as double as it was. This would reduce unnecessary memory usage of mqttSensors for all the strings.

Reply:
The problem is line 330 of `main.cpp`:
`std::map<const char*, std::function<char const*()>, cmp_str> mqttSensors = {};`

The map is defined so that pair.second is a function, which returns a `char const*`. I would need to find a way, to let pair.second return either a `double` or a `char const*` or in the future any other datatype, which might additionally be included in the map.

I do not have another solution for it at the moment, but I am happy for any suggestion.